src/EventStore.Core/Services/Storage/StorageWriterService.cs

```diff
@@ -571,6 +571,10 @@ record = new PrepareLogRecord(newPos,
 				}
 			}
 
+			if (prepare.EventType == SystemEventTypes.ScavengePoint) {
```

Comment:
As a safety measure, should we also verify if stream id = $scavengePoints?
otherwise someone could accidentally (or deliberately) write an event with type `$scavengePoint` which would complete the chunk

Reply:
yeah good idea, i'll make the scavengepointsource more picky about the event type too